# Preview of 12SL Features

## 1. Import libraries

In [4]:
import pandas as pd

## 2. Set file path

In [5]:
plik_12sl = "../data/ptb-xl-plus/12sl_features.csv"

## 3. Load the dataset

In [8]:
df_12sl = pd.read_csv(plik_12sl)
df_12sl.head()

,P_Area_I,P_PeakTime_I,Q_Area_I,Q_PeakTime_I,R_Area_I,R_PeakTime_I,S_Area_I,S_PeakTime_I,QRS_Balance_I,T_Area_I,...,T+_Dur_aVF,T-_Dur_aVF,T+_Amp_aVF,T-_Amp_aVF,T_Morph_aVF,T_DurFull_aVF,P_Dur_Global,P_Found_Global,HR__Global,P_Term_V1
0,0.264,64.0,0.000,0.0,0.737,30.0,0.000,0.0,629.0,0.819,...,208.0,0.0,0.151,0.000,1,208.0,112.0,1,64.0,0.000
1,-0.047,6.0,0.015,8.0,0.794,36.0,1.246,76.0,205.0,2.206,...,0.0,127.0,0.000,-0.078,-1,127.0,66.0,1,75.0,0.000
2,0.000,0.0,0.000,0.0,0.792,38.0,1.389,74.0,141.0,1.155,...,0.0,66.0,0.000,-0.078,-1,66.0,NaN,0,169.0,0.000
3,0.196,74.0,0.000,0.0,0.947,56.0,0.000,0.0,439.0,-0.339,...,0.0,0.0,0.000,0.000,-1,0.0,114.0,1,79.0,0.000
4,0.287,102.0,0.000,0.0,1.021,36.0,0.000,0.0,937.0,0.756,...,104.0,66.0,39.000,-0.034,-2,170.0,136.0,1,92.0,6.612


## 4. Data shape

In [11]:
df_12sl.shape

(21799, 783)

The dataset contains **21,799 rows** and **783 columns**.

- Each row represents one ECG record.
- Each column represents either an identifier or a feature extracted from the ECG signal.

In [12]:
df_12sl.columns.tolist()

['P_Area_I',
 'P_PeakTime_I',
 'Q_Area_I',
 'Q_PeakTime_I',
 'R_Area_I',
 'R_PeakTime_I',
 'S_Area_I',
 'S_PeakTime_I',
 'QRS_Balance_I',
 'T_Area_I',
 'T_PeakTime_I',
 'QRS_Area_I',
 'P_Area_II',
 'P_PeakTime_II',
 'Q_Area_II',
 'Q_PeakTime_II',
 'R_Area_II',
 'R_PeakTime_II',
 'S_Area_II',
 'S_PeakTime_II',
 'QRS_Balance_II',
 'T_Area_II',
 'T_PeakTime_II',
 'QRS_Area_II',
 'P_Area_V1',
 'P_PeakTime_V1',
 'Q_Area_V1',
 'Q_PeakTime_V1',
 'R_Area_V1',
 'R_PeakTime_V1',
 'S_Area_V1',
 'S_PeakTime_V1',
 'QRS_Balance_V1',
 'T_Area_V1',
 'T_PeakTime_V1',
 'QRS_Area_V1',
 'P_Area_V2',
 'P_PeakTime_V2',
 'Q_Area_V2',
 'Q_PeakTime_V2',
 'R_Area_V2',
 'R_PeakTime_V2',
 'S_Area_V2',
 'S_PeakTime_V2',
 'QRS_Balance_V2',
 'T_Area_V2',
 'T_PeakTime_V2',
 'QRS_Area_V2',
 'P_Area_V3',
 'P_PeakTime_V3',
 'Q_Area_V3',
 'Q_PeakTime_V3',
 'R_Area_V3',
 'R_PeakTime_V3',
 'S_Area_V3',
 'S_PeakTime_V3',
 'QRS_Balance_V3',
 'T_Area_V3',
 'T_PeakTime_V3',
 'QRS_Area_V3',
 'P_Area_V4',
 'P_PeakTime_V4',
 'Q_A

The columns are mainly ECG morphology features.  
Their names usually contain the ECG wave or segment, the measured property, and the lead name.

Example: 

`P_Area_I` means the area of the P wave in lead I.  
`R_PeakTime_II` means the peak time of the R wave in lead II.

In [13]:
df_12sl.info()

<class 'pandas.DataFrame'>
RangeIndex: 21799 entries, 0 to 21798
Columns: 783 entries, P_Area_I to P_Term_V1
dtypes: float64(757), int64(26)
memory usage: 130.2 MB


## 5. Missing values analysis

In [14]:
df_12sl.isna().sum().sort_values(ascending=False).head(30)

P_On_Global             2032
P_Off_Global            2032
P_Dur_Global            2032
PR_Int_Global           2013
P_AxisFront_Global      1994
HR_Atrial_Global        1582
T_AxisFront_Global         4
R_AxisFrontal_Global       1
QRS_Balance_I              0
T_Area_I                   0
T_PeakTime_I               0
QRS_Area_I                 0
P_Area_II                  0
P_PeakTime_II              0
Q_Area_II                  0
T+_Amp_aVR                 0
S_Area_V1                  0
P_DurFull_aVF              0
P_PeakTime_I               0
R_Area_II                  0
R_PeakTime_II              0
S_Area_II                  0
S_PeakTime_II              0
QRS_Balance_II             0
T_Area_II                  0
T_PeakTime_II              0
QRS_Area_II                0
P_Area_V1                  0
P_PeakTime_V1              0
Q_Area_V1                  0
dtype: int64

Most columns do not contain missing values.  
The largest number of missing values appears in global P-wave features, which may mean that the P wave was not detected in some ECG records.